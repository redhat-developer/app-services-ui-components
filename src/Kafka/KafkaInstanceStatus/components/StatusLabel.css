.mas-c-status.pf-l-split.pf-m-gutter > *:not(:last-child) {
  --pf-l-split--m-gutter--MarginRight: var(--pf-global--spacer--sm);
}

.mas-m-ready {
  font-size: var(--pf-global--icon--FontSize--md);
  color: var(--pf-global--success-color--100);
}

.mas-m-degraded {
  font-size: var(--pf-global--icon--FontSize--md);
  color: var(--pf-global--warning-color--100);
}

.mas-m-deleting {
  color: var(--pf-global--Color--200);
}
