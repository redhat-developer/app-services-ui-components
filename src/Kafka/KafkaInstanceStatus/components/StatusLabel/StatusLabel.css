.appservices_kafka-instance-status_icon--completed {
  color: var(--pf-global--success-color--100);
}

.appservices_kafka-instance-status_icon--failed {
  color: var(--pf-global--danger-color--100);
}
