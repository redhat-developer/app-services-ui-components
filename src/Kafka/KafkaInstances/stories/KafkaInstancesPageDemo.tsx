import { KafkaInstancesPage } from "../KafkaInstancesPage";
import { useInstances } from "./support";

export const KafkaInstancesPageDemo = () => {
  const { instances } = useInstances();

  return (
    <KafkaInstancesPage
      instances={instances}
      itemCount={instances?.length}
      page={1}
      getUrlForInstance={(row) => `/streams/${row.id}`}
      onChangeOwner={() => {}}
      onClickConnectionTabLink={() => {}}
      onClickSupportLink={() => {}}
      onConnection={() => {}}
      onCreate={() => {}}
      onDelete={() => {}}
      onDetails={() => {}}
      onPageChange={() => {}}
      onQuickstartGuide={() => {}}
    />
  );
};

export default KafkaInstancesPageDemo;
