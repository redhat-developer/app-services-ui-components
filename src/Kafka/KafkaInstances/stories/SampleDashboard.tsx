export default () => <div>Dashboard</div>;
