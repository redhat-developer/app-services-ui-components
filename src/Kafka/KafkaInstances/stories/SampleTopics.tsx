export default () => <div>Topics</div>;
