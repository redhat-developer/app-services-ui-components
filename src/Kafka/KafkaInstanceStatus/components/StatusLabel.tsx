import {
  Alert,
  Button,
  Flex,
  FlexItem,
  HelperText,
  HelperTextItem,
  Spinner,
  Split,
  SplitItem,
} from "@patternfly/react-core";
import {
  CheckCircleIcon,
  ExclamationTriangleIcon,
} from "@patternfly/react-icons";
import { forwardRef } from "react";
import { useTranslation } from "react-i18next";
import type { Status } from "../../types";
import {
  CreatingStatuses,
  DegradedStatuses,
  DeletingStatuses,
  ReadyStatuses,
} from "../../types";
import "./StatusLabel.css";

type StatusLabelProps = {
  value: Status;
  showWarning?: boolean;
  showError?: boolean;
  withPopover?: boolean;
};

/**
 *<strong> Introduction </strong>
 *These are the various statuses that can show in the Kafka instances table under the "Status" header.
 */
export const StatusLabel = forwardRef<HTMLButtonElement, StatusLabelProps>(
  (
    { value, showWarning = false, showError = false, withPopover = false },
    ref
  ) => {
    const { t } = useTranslation("kafka");

    const buttonVariant = withPopover ? "link" : "plain";

    switch (true) {
      case ReadyStatuses.includes(value):
        return (
          <div>
            <Split hasGutter className="mas-c-status">
              <SplitItem>
                <CheckCircleIcon className="mas-m-ready" />
              </SplitItem>
              <SplitItem>{t("statuses.ready")}</SplitItem>
            </Split>
          </div>
        );

      case CreatingStatuses.includes(value):
        switch (true) {
          case showWarning:
            return (
              <div>
                <Split hasGutter className="mas-c-status">
                  <SplitItem>
                    <Spinner size="md" />
                  </SplitItem>
                  <SplitItem>
                    <Button ref={ref} variant={buttonVariant} isInline>
                      {t("statuses.creating")}
                    </Button>
                  </SplitItem>
                </Split>
                <Alert
                  variant="warning"
                  isInline
                  isPlain
                  title="This is taking longer than expected."
                />
              </div>
            );
          case showError:
            return (
              <div>
                <Split hasGutter className="mas-c-status">
                  <SplitItem>
                    <Spinner size="md" />
                  </SplitItem>
                  <SplitItem>
                    <Button ref={ref} variant={buttonVariant} isInline>
                      {t("statuses.creating")}
                    </Button>
                  </SplitItem>
                </Split>
                <Alert
                  variant="danger"
                  isInline
                  isPlain
                  title={t("status_warning_or_error_title")}
                />
              </div>
            );
          default:
            return (
              <div>
                <Split hasGutter className="mas-c-status">
                  <SplitItem>
                    <Spinner size="md" />
                  </SplitItem>
                  <SplitItem>
                    <Button ref={ref} variant={buttonVariant} isInline>
                      {t("statuses.creating")}
                    </Button>
                    <Flex>
                      <FlexItem>
                        <HelperText>
                          <HelperTextItem variant="indeterminate">
                            {t("status_created_shortly_help")}
                          </HelperTextItem>
                        </HelperText>
                      </FlexItem>
                    </Flex>
                  </SplitItem>
                </Split>
              </div>
            );
        }

      case DegradedStatuses.includes(value):
        return (
          <div>
            <Split hasGutter className="mas-c-status">
              <SplitItem>
                <ExclamationTriangleIcon className="mas-m-degraded" />
              </SplitItem>
              <SplitItem>{t("statuses.degraded")}</SplitItem>
            </Split>
          </div>
        );

      case DeletingStatuses.includes(value):
        return (
          <div>
            <p className="mas-m-deleting"> {t("statuses.deleting")}</p>
          </div>
        );
      default:
        return null;
    }
  }
);
