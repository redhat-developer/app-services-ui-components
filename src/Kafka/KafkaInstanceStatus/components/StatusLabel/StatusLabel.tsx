import { Flex, FlexItem, Spinner } from "@patternfly/react-core";
import CheckCircleIcon from "@patternfly/react-icons/dist/js/icons/check-circle-icon";
import ExclamationCircleIcon from "@patternfly/react-icons/dist/js/icons/exclamation-circle-icon";
import PendingIcon from "@patternfly/react-icons/dist/js/icons/pending-icon";
import React, {
  FunctionComponent,
  ReactElement,
  VoidFunctionComponent,
} from "react";
import { useTranslation } from "react-i18next";
import { KafkaStatus } from "../../types";
import "./StatusLabel.css";

type StatusLabelProps = {
  value: KafkaStatus;
  withPopover?: boolean;
};

export const StatusLabel: VoidFunctionComponent<StatusLabelProps> = ({
  value,
  withPopover,
}) => {
  const { t } = useTranslation(["create-kafka-instance"]);
  const translations: { [status in KafkaStatus]: string } = {
    ready: t("statuses.ready"),
    failed: t("statuses.failed"),
    accepted: t("statuses.accepted"),
    provisioning: t("statuses.provisioning"),
    preparing: t("statuses.preparing"),
    deprovision: t("statuses.deprovision"),
    deleting: t("statuses.deleting"),
  };
  const statusCopy = translations[value];
  const statusElement = withPopover ? <a>{statusCopy}</a> : statusCopy;

  switch (value) {
    case "ready":
      return (
        <StatusWithIcon
          icon={
            <CheckCircleIcon className="appservices_kafka-instance-status_icon--completed" />
          }
        >
          {statusElement}
        </StatusWithIcon>
      );
    case "failed":
      return (
        <StatusWithIcon
          icon={
            <ExclamationCircleIcon className="appservices_kafka-instance-status_icon--failed" />
          }
        >
          {statusElement}
        </StatusWithIcon>
      );
    case "accepted":
      return (
        <StatusWithIcon icon={<PendingIcon />}>{statusElement}</StatusWithIcon>
      );
    case "provisioning":
    case "preparing":
      return (
        <StatusWithIcon
          icon={<Spinner size="md" aria-valuetext="Creation in progress" />}
        >
          {statusElement}
        </StatusWithIcon>
      );
    case "deprovision":
    case "deleting":
      return <StatusWithIcon>{statusElement}</StatusWithIcon>;
    default:
      return (
        <StatusWithIcon icon={<PendingIcon />}>{statusElement}</StatusWithIcon>
      );
  }
};

const StatusWithIcon: FunctionComponent<{ icon?: ReactElement }> = ({
  icon,
  children,
}) => (
  <Flex display={{ default: "inlineFlex" }}>
    {icon && <FlexItem spacer={{ default: "spacerSm" }}>{icon}</FlexItem>}
    <FlexItem>{children}</FlexItem>
  </Flex>
);
