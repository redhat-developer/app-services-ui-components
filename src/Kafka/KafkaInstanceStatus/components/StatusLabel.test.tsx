import { composeStories } from "@storybook/testing-react";
import { render, waitForI18n } from "../../../test-utils";
import * as stories from "./StatusLabel.stories";

const testCases = Object.values(composeStories(stories)).map((Story) => [
  Story.storyName!,
  Story,
]);

// Batch snapshot testing
test.each(testCases)("Renders %s story", async (_storyName, Story) => {
  const tree = render(<Story />);
  await waitForI18n(tree);
  expect(tree.baseElement).toMatchSnapshot();
});
