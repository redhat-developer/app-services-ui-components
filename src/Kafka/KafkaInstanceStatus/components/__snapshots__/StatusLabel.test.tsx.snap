// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Accepted story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-1"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
          <div
            class="pf-l-flex"
          >
            <div
              class=""
            >
              <div
                class="pf-c-helper-text"
              >
                <div
                  class="pf-c-helper-text__item pf-m-indeterminate"
                >
                  <span
                    class="pf-c-helper-text__item-text"
                  >
                    This will be ready shortly.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders AllStatutesAtOnce story 1`] = `
<body>
  <div>
    <pre
      style=""
    >
      ready
    </pre>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <svg
            aria-hidden="true"
            class="mas-m-ready"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            />
          </svg>
        </div>
        <div
          class="pf-l-split__item"
        >
          Ready
        </div>
      </div>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      accepted
    </pre>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-2"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
          <div
            class="pf-l-flex"
          >
            <div
              class=""
            >
              <div
                class="pf-c-helper-text"
              >
                <div
                  class="pf-c-helper-text__item pf-m-indeterminate"
                >
                  <span
                    class="pf-c-helper-text__item-text"
                  >
                    This will be ready shortly.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      provisioning
    </pre>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-3"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
          <div
            class="pf-l-flex"
          >
            <div
              class=""
            >
              <div
                class="pf-c-helper-text"
              >
                <div
                  class="pf-c-helper-text__item pf-m-indeterminate"
                >
                  <span
                    class="pf-c-helper-text__item-text"
                  >
                    This will be ready shortly.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      preparing
    </pre>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-4"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
          <div
            class="pf-l-flex"
          >
            <div
              class=""
            >
              <div
                class="pf-c-helper-text"
              >
                <div
                  class="pf-c-helper-text__item pf-m-indeterminate"
                >
                  <span
                    class="pf-c-helper-text__item-text"
                  >
                    This will be ready shortly.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      accepted
    </pre>
     over 15 minutes
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-5"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
        </div>
      </div>
      <div
        aria-label="Warning Alert"
        class="pf-c-alert pf-m-inline pf-m-plain pf-m-warning"
        data-ouia-component-id="OUIA-Generated-Alert-warning-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Warning alert:
          </span>
          This is taking longer than expected.
        </h4>
        
      </div>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      accepted
    </pre>
     over 30 minutes
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-6"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
        </div>
      </div>
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-inline pf-m-plain pf-m-danger"
        data-ouia-component-id="OUIA-Generated-Alert-danger-1"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          This is taking longer than expected.
        </h4>
        
      </div>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      deleting
    </pre>
    <div>
      <p
        class="mas-m-deleting"
      >
         
        Deleting
      </p>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      deprovision
    </pre>
    <div>
      <p
        class="mas-m-deleting"
      >
         
        Deleting
      </p>
    </div>
    <br
      style=""
    />
    <pre
      style=""
    >
      degraded
    </pre>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <svg
            aria-hidden="true"
            class="mas-m-degraded"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <div
          class="pf-l-split__item"
        >
          Degraded
        </div>
      </div>
    </div>
    <br
      style=""
    />
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders CreatingOver15Minutes story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-7"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
        </div>
      </div>
      <div
        aria-label="Warning Alert"
        class="pf-c-alert pf-m-inline pf-m-plain pf-m-warning"
        data-ouia-component-id="OUIA-Generated-Alert-warning-2"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Warning alert:
          </span>
          This is taking longer than expected.
        </h4>
        
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders CreatingOver30Minutes story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-8"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
        </div>
      </div>
      <div
        aria-label="Danger Alert"
        class="pf-c-alert pf-m-inline pf-m-plain pf-m-danger"
        data-ouia-component-id="OUIA-Generated-Alert-danger-2"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Danger alert:
          </span>
          This is taking longer than expected.
        </h4>
        
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders Degraded story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <svg
            aria-hidden="true"
            class="mas-m-degraded"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 576 512"
            width="1em"
          >
            <path
              d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            />
          </svg>
        </div>
        <div
          class="pf-l-split__item"
        >
          Degraded
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders Deleting story 1`] = `
<body>
  <div>
    <div>
      <p
        class="mas-m-deleting"
      >
         
        Deleting
      </p>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders Deprovision story 1`] = `
<body>
  <div>
    <div>
      <p
        class="mas-m-deleting"
      >
         
        Deleting
      </p>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders Preparing story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-9"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
          <div
            class="pf-l-flex"
          >
            <div
              class=""
            >
              <div
                class="pf-c-helper-text"
              >
                <div
                  class="pf-c-helper-text__item pf-m-indeterminate"
                >
                  <span
                    class="pf-c-helper-text__item-text"
                  >
                    This will be ready shortly.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders Provisioning story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <span
            aria-label="Contents"
            aria-valuetext="Loading..."
            class="pf-c-spinner pf-m-md"
            role="progressbar"
          >
            <span
              class="pf-c-spinner__clipper"
            />
            <span
              class="pf-c-spinner__lead-ball"
            />
            <span
              class="pf-c-spinner__tail-ball"
            />
          </span>
        </div>
        <div
          class="pf-l-split__item"
        >
          <button
            aria-disabled="false"
            class="pf-c-button pf-m-plain"
            data-ouia-component-id="OUIA-Generated-Button-plain-10"
            data-ouia-component-type="PF4/Button"
            data-ouia-safe="true"
            type="button"
          >
            Creating
          </button>
          <div
            class="pf-l-flex"
          >
            <div
              class=""
            >
              <div
                class="pf-c-helper-text"
              >
                <div
                  class="pf-c-helper-text__item pf-m-indeterminate"
                >
                  <span
                    class="pf-c-helper-text__item-text"
                  >
                    This will be ready shortly.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;

exports[`Renders Ready story 1`] = `
<body>
  <div>
    <div>
      <div
        class="pf-l-split pf-m-gutter mas-c-status"
      >
        <div
          class="pf-l-split__item"
        >
          <svg
            aria-hidden="true"
            class="mas-m-ready"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            />
          </svg>
        </div>
        <div
          class="pf-l-split__item"
        >
          Ready
        </div>
      </div>
    </div>
  </div>
  <div>
    <ul
      aria-atomic="false"
      aria-live="polite"
      class="pf-c-alert-group pf-m-toast"
    />
  </div>
</body>
`;
